lib/runtime/procs/source.js

```diff
@@ -3,16 +3,74 @@
 //
 
 var juttle_utils = require('../juttle-utils');
+var values = require('../values');
 var _ = require('underscore');
 
 var base = require('./base');
 
 var INFO = {
     type: 'source',
-    options: {}
+    options: {
+        from: {},
+        to: {},
+        last: {}
+    }
 };
 
 var source = base.extend({
+    initialize: function(options, params) {
+        this.handleTimeOptions(options);
+    },
+
+    // Process the common time options (from / to / last) to ensure they are all
+    // the correct types and to convert `last` into the corresponding from/to
+    // combinations.
+    //
+    // Stashes the time bounds in `this.from` and `this.to`.
+    handleTimeOptions: function(options) {
+        if (options.from) {
+            if (! options.from.moment) {
+                throw this.compile_error('RT-MOMENT-ERROR', {
```

Comment:
`function throwMomentError(option) { throw self.compile_error('RT-MOMENT-ERROR', {option: option, value: values.inspect(options[option])})}` could DRY this up a bit.